test/app.test.ts

```diff
@@ -163,6 +163,45 @@ test('app with chart dependencies via custom constructs', () => {
 
 });
 
+test('app with nested charts will deduplicate api objects', () => {
```

Comment:
Can you also a test case that explicitly simulates the scenario reported in the issue? This stuff is hard to reason about and I want to make sure we address the initial report.

Reply:
Added.